### PR TITLE
CI: Separate lint and add pypy

### PR DESCRIPTION
**Pull request by xuhdev**

### CODE REVIEW

File: requirements-test.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Dependencies for runtime test
+coverage
+pytest
+tox
```
Comment (edwardleardi) [inline requirements-test.txt]: Where should I place `sphinx` and other docs related packages? In this file, or maybe a separate file like `requirements-docs.txt`?

File: tox.ini [hunk 2]
```diff
@@ -1,14 +1,25 @@
 [tox]
-envlist = py{36,37,38,39}
+envlist = py{36,37,38,39},pypy3,lint
```
Comment (edwardleardi) [inline tox.ini]:
```suggestion
envlist = py{36,37,38,39,py3},lint
```
I think you can do this

File: .travis.yml [hunk 3]
```diff
@@ -14,26 +14,27 @@
 # limitations under the License.
 #
 
+# The prebuilt pypy packages at https://antocuni.github.io/ don't work with the older default dist
```
Comment (edwardleardi) [inline .travis.yml]:
```suggestion
# The prebuilt pypy packages at https://antocuni.github.io/pypy-wheels/ don't work with the older default dist
```
https://antocuni.github.io/ not a valid url, do you want to use https://antocuni.github.io/pypy-wheels/ instead?

File: requirements-test.txt (hunk 1 above)
Comment (xuhdev) [reply requirements-test.txt]: I didn't consider this. I've updated the requirement structure and now it should be more organized. You can add doc stuff in `requirements/doc.txt` (looks like we always use the singular form?)

Comment (edwardleardi) [reply requirements-test.txt]: thanks

**Review by edwardleardi — APPROVED** [review]
LGTM

**Pull request merged**